libs/wingsdk/src/platform/parameter-registrar.ts

```diff
@@ -0,0 +1,163 @@
+import Ajv from "ajv";
+import { Construct } from "constructs";
+import { loadPlatformSpecificValues } from "./util";
+
+/**
+ * Parameter Registrar
+ *
+ * This class is used to register and lookup parameter values.
+ */
+export class ParameterRegistrar extends Construct {
+  /** Cache for parameter lookups */
+  private parameterValueByPath: { [key: string]: any } = {};
+  /** List of all registered parameter schemas */
+  private parameterSchemas: any[] = [];
+
+  /** @internal */
+  public readonly _rawParameters: { [key: string]: any } =
+    loadPlatformSpecificValues();
+
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+  }
+
+  /**
+   * Retrieve a parameter value by its path
+   *
+   * @param path the path of the parameter
+   * @returns the value of the parameter
+   */
+  public getParameterValue(path: string): any {
+    if (this.parameterValueByPath[path] === undefined) {
+      // attempt to read the value from the raw parameters, then cache it
+      this.parameterValueByPath[path] = resolveValueFromPath(
+        this._rawParameters,
+        path
+      );
+    }
+
+    return this.parameterValueByPath[path];
+  }
```

Comment:
What happens if the parameter isn't found at the path (is this possible)?

Reply:
So in this case it would return undefined, which if the parameter was an optional parameter than that should be expected, however if its required then during preSynth there would be a missing required parameter error